docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -234,316 +317,405 @@ async function run() {
+
+The `run` function then sets up a connection to your Temporal Server, invokes your Workflow, passes in an argument for the `name`, and assigns the Workflow a unique identifier with Nanoid. The client dispatches the Workflow on the same `hello-world` Task Queue that the Worker is polling on.
+
+:::note Specify a Workflow Id
+
+A Workflow ID is unique in a Namespace and identifies a Workflow Execution. Using an identifier that reflects some business process or entity is a good practice. For example, you might use a customer identifier as part of the Workflow Id if you run one Workflow per customer. This would make it easier to find all of the Workflow Executions related to that customer later.
+
+In this tutorial you're generating a UUID and appending it to a string that identifies the Workflow.
+:::
+
+With the script created, open a new terminal and switch to the project directory:
 
 ```command
-npm test
+cd ip-geo-ts
+```
+
+Now run the following command to run the Workflow using the client program you wrote:
+
+```command
+npm run workflow Brian
 ```
 
-You'll see output similar to the following from your test run indicating that the test was successful:
+You'll see the following output:
+
+```output
+> temporal-hello-world@0.1.0 workflow
+> ts-node src/client.ts Brian
 
+Hello, Brian. Your IP is 198.51.100.23 and your location is Seattle,  Washington, United States
 ```
-> temporal-hello-world@0.1.0 test
-> mocha --exit --require ts-node/register --require source-map-support/register src/mocha/*.test.ts
 
-  greet activity
-    ✓ successfully greets the user
+Your Temporal application works. Now review it in the Temporal Web UI.
+
+## Exploring your application in the Web UI
+
+The Temporal Web UI gives you insights into your Workflow's execution. Open the Temporal Web UI by visiting `http://localhost:8233` and click on your completed Workflow to view the execution history. You'll see results similar to the following image:
+
+![The results of the Workflow](images/overview.png)
+
+You'll see the dates the Workflow Exeuction ran, how long it took to run, the input to the Workflow, and the result.
+
+Below that, you see the Event History, detailing the entire flow, including the inputs and outputs of the Activity Executions:
+
+![The Workflow History](images/history.png)
+
+The most recent event is at the top, so read the history from the bottom up to see each step in the process. Using this history, you can see exactly how your Workflow executed and pinpoint any places things might have gone wrong.
+
+Temporal stores the results of each Activity in this history, as you can see im the image. If there was a system crash between the `getIP` and `getLocationInfo` Activity Executions, a new Worker would re-run the Workflow, but would use the previous Event History to reconstruct the Workflow's state. Instead of re-running the `getIP` function, it would use the previous run's value and continue on.
```

Comment:
I would re-emphasize why this is important. Consider adding something like at the end:

This is important in the case of failure because it prevents duplicate external calls or side effects, such as calling the same API multiple times or generating inconsistent results. By relying on the stored Event History, Temporal ensures that the Workflow can recover seamlessly, maintaining reliability and consistency even after a crash.

Reply:
Isn't this covered in the next paragraph?

Reply:
I like emphasis lol. It introduces the concept here and then re-emphasizes as you get to the next part. Up to you though!